app/api/services/case_study.py

```diff
@@ -0,0 +1,92 @@
+from app import db
+from app.api.helpers import Service
+from app.models import CaseStudy, Domain
+
+from sqlalchemy import and_, func, or_
+
+
+class CaseStudyService(Service):
+    __model__ = CaseStudy
+
+    def __init__(self, *args, **kwargs):
+        super(CaseStudyService, self).__init__(*args, **kwargs)
+
+    def get_approved_case_studies_by_supplier_code(self, supplier_code, domain_id):
+        domain_name = (
+            db
+            .session
+            .query(Domain.name)
+            .filter(Domain.id == domain_id)
+            .subquery()
+        )
+
+        cs_id = (
+            db.session.query(
+                CaseStudy.id.label('case_study_id')
+            )
+            .filter(
+                CaseStudy.supplier_code == supplier_code,
+                CaseStudy.status == 'approved',
+                CaseStudy.data['service'].astext == domain_name.c.name,
+            )
+            .subquery()
+        )
+
+        case_study = (
+            db.session.query(
+                domain_name.c.name,
+                func.json_object_agg(cs_id.c.case_study_id,
+                                     func.json_build_object('approach',
+                                                            CaseStudy.data['client'].label('approach'),
+                                                            'client',
+                                                            CaseStudy.data['client'].label('client'),
+                                                            'opportunity',
```

Comment:
Is it easier to read if we add new lines after the opening parenthesis and reduce the indentation?
```python
                func.json_object_agg(
                    cs_id.c.case_study_id,
                    func.json_build_object(
                        'approach',
                        CaseStudy.data['client'].label('approach'),
                        'client',
                        CaseStudy.data['client'].label('client'),
```